AndroidAnnotations/androidannotations/src/main/java/org/androidannotations/handler/ResHandler.java

```diff
@@ -100,4 +115,42 @@ public void process(Element element, EComponentHolder holder) {
 	protected boolean hasContextCompatInClasspath() {
 		return processingEnvironment().getElementUtils().getTypeElement(CanonicalNameConstants.CONTEXT_COMPAT) != null;
 	}
+	
+	protected boolean hasContextWithGetDrawableMethod() {
+		return androidManifest.getMinSdkVersion() >= MIN_SDK_WITH_CONTEXT_GET_DRAWABLE;
+	}
+
+	protected boolean hasGetDrawableInContext() {
+		TypeElement context = processingEnvironment().getElementUtils().getTypeElement(CanonicalNameConstants.CONTEXT);
+
+		return hasGetDrawable(context);
+	}
+
+	private void createCallWithIfGuard(EComponentHolder holder, JFieldRef ref, JBlock methodBody, JFieldRef idRef) {
+		JVar resourcesRef = holder.getResourcesRef();
+		JConditional guardIf = methodBody._if(holder.classes().BUILD_VERSION.staticRef("SDK_INT").gte(holder.classes().BUILD_VERSION_CODES.staticRef("LOLLIPOP")));
+		JBlock ifBlock = guardIf._then();
+		ifBlock.assign(ref, holder.getContextRef().invoke("getDrawable").arg(idRef));
+
+		JBlock elseBlock = guardIf._else();
+		elseBlock.assign(ref, resourcesRef.invoke("getDrawable").arg(idRef));
+	}
+
+
+	private boolean hasGetDrawable(TypeElement type) {
+		if (type == null) {
+			return false;
+		}
+
+		for (Element element : type.getEnclosedElements()) {
```

Comment:
You can use `ElementFilter.methodsIn(type.getEnclosedElements())` to remove the need of the first `if`.